pkg/oam/util/helper.go

```diff
@@ -633,6 +647,138 @@ func RawExtension2AppConfig(raw runtime.RawExtension) (*v1alpha2.ApplicationConf
 	return ac, nil
 }
 
+// RawExtension2Component converts runtime.RawExtention to Component
+func RawExtension2Component(raw runtime.RawExtension) (*v1alpha2.Component, error) {
+	c := &v1alpha2.Component{}
+	b, err := raw.MarshalJSON()
+	if err != nil {
+		return nil, err
+	}
+	if err := json.Unmarshal(b, c); err != nil {
+		return nil, err
+	}
+	return c, nil
+}
+
+// ExtractComponentName will extract the componentName from a revisionName
+func ExtractComponentName(revisionName string) string {
+	splits := strings.Split(revisionName, "-")
+	return strings.Join(splits[0:len(splits)-1], "-")
+}
+
+// AppConfig2ComponentManifests convert AppConfig and Components to a slice of ComponentManifest.
+func AppConfig2ComponentManifests(acRaw runtime.RawExtension, comps []common.RawComponent) ([]*oamtypes.ComponentManifest, error) {
```

Comment:
we should write unit test to cover the error case

Reply:
yes, WIP.